src/TrackingOptOut.php

```diff
@@ -106,7 +106,7 @@ public static function renderSettings() {
   }
 
   public static function enqueueScriptsFirstHead() {
-    wp_enqueue_script(Plugin::PREFIX . '-google_analytics_opt_out', Plugin::getBaseUrl() . '/dist/scripts/google-analytics-opt-out.js');
+    wp_enqueue_script(Plugin::PREFIX . '-google_analytics_opt_out', Plugin::getBaseUrl() . '/dist/scripts/google-analytics-opt-out.js', FALSE, Plugin::getGitRef());
```

Comment:
The manual approach is prone to errors and mistakes and won't catch all asset files needing a version. Any chance to apply this on a more global scope for all scripts/styles that do not state a version yet?

Are the enqueue API functions invoking any hooks? (or alternatively the rendering functions)

Reply:
I do not agree. Some styles/scripts will depend on the git version (project specific theme and custom module) others on the plugin version (basically all plugins), others in the WordPress version. Handle it with a generic filter on render or process will require to handle this logic in a centralized place (or implement it in different places for each case, so again not centralized). IMHO makes more sense that each plugin or theme is aware and responsible to handle the cache invalidation of its styles and scripts.

Reply:
Hence the suggestion to only apply it automatically to all assets that do not supply a version yet. The current Git commit hash always works as a safe fallback to guarantee that all assets have one. Any other specialized version hash only improves the invalidation frequency (to not invalidate automatically with every deployment).

Reply:
ok, I will rework it.